### PR TITLE
tools: make cmuxTests wiring deterministic

### PR DESCRIPTION
**Pull request by teamleaderleo**
## Reviewer summary

Makes cmuxTests wiring repeatable and visible. New tests are added to the right Xcode target, incomplete entries are repaired, and --check catches drift before CI can report that zero tests ran.

## What changed

Add a repo-owned `./scripts/sync-test-wiring` authoring path for direct `cmuxTests/*.swift` files so a newly created test cannot silently miss the Xcode unit-test target.

- deterministically reconciles PBXFileReference, PBXBuildFile, the `cmuxTests` group child, and the `cmuxTests` Sources phase
- preserves existing identifiers where possible and derives stable SHA-256-based 24-character identifiers for new objects
- repairs incomplete same-target wiring and same-target duplicates
- rejects membership in `cmux`, `cmuxUITests`, or any other Sources target with an explicit diagnostic
- treats only direct `cmuxTests/*.swift` entries as managed, preserving intentional nested/SOURCE_ROOT references
- supports `--check` and `--dry-run`; reruns are byte-for-byte idempotent
- keeps `lint-pbxproj-test-wiring.sh` as the defensive Sources-phase guard

The live project also had three inconsistencies that the stronger sync model exposed: two test Sources entries were missing their PBXBuildFile objects, and one test had duplicate same-target file/build membership. This PR repairs those entries without changing target scope.

## Tests

- `python3 -m py_compile scripts/sync_test_wiring.py tests/test_sync_test_wiring.py`
- `bash -n scripts/sync-test-wiring scripts/lint-pbxproj-test-wiring.sh`
- `python3 tests/test_sync_test_wiring.py`: 14 fixture tests pass
- verified current project wiring across 974 direct `cmuxTests/*.swift` files: zero missing/duplicate/wrong-target/stale entries after reconciliation
- fixture coverage proves add, idempotent rerun, incomplete repair, wrong-target rejection, deletion detection/pruning, deterministic output, pbxproj normalization compatibility, and agreement with the existing lint
- review hardening covers OpenStep quoting for `+` filenames, multiline file references, stale display-comment repair by object identity, and duplicate same-path references in foreign targets

Tact #79, Lane R.

Replaces https://github.com/manaflow-ai/cmux/pull/13053 (same commits, head branch moved into the org so it gets the build cache and can be kept current with main).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
<a href="https://app.blacksmith.sh/manaflow-ai/codesmith/cmux/pr/13216?ref=codesmith_pr_footer"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with [code]smith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1792497044&installation_model_id=21920&pr_number=13216&ref=codesmith_pr_footer&repository=manaflow-ai%2Fcmux&return_to=https%3A%2F%2Fgithub.com%2Fmanaflow-ai%2Fcmux%2Fpull%2F13216&signature=800191d000a94e32b0aa6e6da54bdec8423ea0c64750d6d12b4857124e62f3db"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with [code]smith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith-bot</code> with what you need. Autofix is disabled.</sup>

---
## Summary by cubic
Makes direct `cmuxTests/*.swift` wiring deterministic for Tact #79. Instead of silently skipping files with "Executed 0 tests," `./scripts/sync-test-wiring` now reconciles their Xcode project entries and rejects foreign-target membership with an explicit diagnostic.

- Repairs missing, duplicate, stale, and deleted entries while preserving object identities, neighboring objects, shared sources owned by other groups, nested references, and `SOURCE_ROOT` references.
- `--check` and `--dry-run` are read-only; reruns are idempotent, and output matches `normalize-pbxproj.py` ordering.
- CI runs `sync-test-wiring --check` alongside `lint-pbxproj-test-wiring.sh`, with fixtures covering repairs, deletion, duplicates, target safety, normalization, and stale comments.
- Repairs the live project by adding Sources entries for two cloud notification tests, migrating those tests to current owners, and removing duplicate `CmuxTuiSurfaceProviderRegistryPollingTests` wiring.

<sup>Written for commit 3ae9b563c467cb04378b357da035b6ba57750678. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/manaflow-ai/cmux/pull/13216?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

- **New Features**
  - Added automated synchronization for Swift test files and Xcode project test-target wiring.
  - Added validation and preview modes to check for inconsistencies without modifying files.
  - Added clearer guidance when test files are missing from the test target.

- **Documentation**
  - Updated testing guidance to recommend the synchronization workflow after adding, renaming, or deleting tests.

- **Tests**
  - Added comprehensive coverage for synchronization, validation, duplicate entries, deleted files, and invalid target membership.
  - Registered new notification-related test files in the test target.

### CODE REVIEW

File: scripts/sync_test_wiring.py [hunk 1]
```diff
@@ -0,0 +1,1207 @@
...
+
+        # Normalize every cmuxTests source membership tied to this file reference,
+        # even when an old Xcode edit left stale display comments behind.
+        sources = _sources_blocks(text)[tests_sources_id]
+        current_entries = _list_entries(sources.text, "files")
+        duplicate_entries = [
+            (entry_id, comment)
+            for entry_id, comment in current_entries
+            if (entry_id in linked_ids or comment == expected_source_comment)
+            and entry_id != canonical_build_id
```
[the remark this answers is not in the document]
Comment (teamleaderleo) [reply scripts/sync_test_wiring.py]: Fixed in 6a9a2bda4cb2f00b41ccc52b93900aee6e6ce0d7, with regression commit 4ce52cb0f7b32cb507c8daa129436529c528e3e6. Existing PBXBuildFile objects are now matched by fileRef identity; a matching display comment is considered only when the build object is genuinely missing. Deleted-source dangling cleanup likewise preserves any existing object with another fileRef. Regressions cover both a present and deleted managed file with a stale-comment SOURCE_ROOT neighbor. The 17-test wiring suite passes at 81925548115fd7045da6cdeda4bfeb620aa82adc; the old implementation produced three expected failures across the two reviewed concerns.

File: scripts/sync_test_wiring.py [hunk 2]
```diff
@@ -0,0 +1,1207 @@
...
+            ):
+                text = _remove_object_line(text, "PBXBuildFile", build.identifier)
+                actions.append(f"removed unused duplicate PBXBuildFile for {filename}")
+
+    # Remove direct cmuxTests file references whose files were deleted. External
+    # SOURCE_ROOT references and nested paths are deliberately outside this tool.
+    disk = set(filenames)
+    file_refs = _parse_file_references(text)
+    group = _find_cmux_tests_group(text)
+    direct_refs_by_name = _direct_group_refs(group, file_refs)
```
[the remark this answers is not in the document]
Comment (teamleaderleo) [reply scripts/sync_test_wiring.py:1006]: Fixed in 6a9a2bda4cb2f00b41ccc52b93900aee6e6ce0d7, with regression commit 4ce52cb0f7b32cb507c8daa129436529c528e3e6. Deleted-file discovery now includes direct references identified by cmuxTests Sources memberships independently of group children. It removes a missing file even when the group child was already removed, while preserving nested/SOURCE_ROOT entries and refusing conflicting target/group ownership. The regression reproduced --check incorrectly passing before the repair. All 17 wiring tests now pass; the existing duplicate-fixture neighbor is explicitly present on disk in 81925548115fd7045da6cdeda4bfeb620aa82adc so preservation is tested independently of deleted-file cleanup. Native CI remains required.

**Comment by teamleaderleo** [comment]
Current-head repair: `848883d803de01ea82cc79a6b3061a6296539e92`. Both P1 findings are repaired with object-identity reconciliation and orphaned Sources membership detection. The expanded suite passes 18 tests. The current real project (1,014 direct test files) also reconciles unchanged, including shared production sources owned by other groups; they are not treated as deleted tests. Upstream main `024562cb7f53be3cb56658efe93b44cbd7354e1b` is integrated. The inherited missing CmuxFoundation imports were repaired from upstream #13236. Both reviewed threads are answered/resolved, and full-ci is enabled for current-head validation; no native test pass is claimed locally.

File: scripts/sync_test_wiring.py [hunk 3]
```diff
@@ -0,0 +1,1236 @@
...
+        if build.identifier in test_build_ids
+    }
+    remaining_groups = _group_memberships(text)
+    for filename, refs in _direct_refs_by_name(file_refs).items():
+        known_ids = {ref.identifier for ref in direct_refs_by_name.get(filename, [])}
+        for ref in refs:
+            if (
+                ref.identifier in test_ref_ids
+                and ref.identifier not in known_ids
+                and not remaining_groups.get(ref.identifier)
```
[the remark this answers is not in the document]
Comment (teamleaderleo) [reply scripts/sync_test_wiring.py:1024]: Rechecked exact head 848883d803de01ea82cc79a6b3061a6296539e92. A group-relative reference owned by Sources is a shared production source, not evidence of a deleted cmuxTests file. The real project deliberately compiles AgentHookNotificationPolicy.swift through reference A76110010000000000000002 in cmux-cli and cmuxTests. Its basename is absent from the direct cmuxTests disk list by design. Removing or rejecting its cmuxTests membership would break that intended wiring (and previously made the actual-project CI guard fail). This tool owns direct cmuxTests group entries and ungrouped orphan test references; it has no historical ownership data to infer that a foreign-owned reference used to be a direct test. Entries still owned by cmuxTests retain the foreign-membership rejection. The existing test_shared_production_source_keeps_its_own_group_owner and deleted-file orphan regression both pass; 18/18 tests pass, and the actual project reconciles unchanged. Resolving this requested deletion as unsafe for the tool’s documented direct-test scope.

File: scripts/sync_test_wiring.py [hunk 4]
```diff
@@ -0,0 +1,1236 @@
...
+            or refs[0].identifier != all_refs[0].identifier
+        ):
+            dirty.append(filename)
+            continue
+        ref = refs[0]
+        group_comments = [
+            comment
+            for entry_id, comment in _list_entries(group.text, "children")
+            if entry_id == ref.identifier
+        ]
```
[the remark this answers is not in the document]
Comment (teamleaderleo) [reply scripts/sync_test_wiring.py]: Addressed in 848883d803de01ea82cc79a6b3061a6296539e92. Lazily index group child comments once per dirty-file check and retain the existing short-circuit parsing order. All 18 focused Python tests pass; full intended CI is requested on the pushed head.

File: scripts/sync_test_wiring.py [hunk 5]
```diff
@@ -0,0 +1,1236 @@
...
+            print("sync-test-wiring: clean")
+        return 0
+
+    if args.check:
+        if changed:
+            print("sync-test-wiring: project wiring is out of sync", file=sys.stderr)
+            for action in result.actions:
+                print(f"  - {action}", file=sys.stderr)
+            print("run ./scripts/sync-test-wiring and commit the pbxproj change", file=sys.stderr)
+            return 1
```
[the remark this answers is not in the document]
Comment (teamleaderleo) [reply scripts/sync_test_wiring.py:1224]: Addressed in 848883d803de01ea82cc79a6b3061a6296539e92. Print an explicit normalization reason when --check detects text changes without wiring actions. All 18 focused Python tests pass; full intended CI is requested on the pushed head.

**Comment by teamleaderleo** [comment]
Exact-head app-host compilation exposed dormant tests newly included by this wiring repair: obsolete CmuxTuiSurfaceProvider.notificationTarget calls and access to private CloudLinkPipe.LineBuffer. Fixed in3ae9b563c467cb04378b357da035b6ba57750678. The attribution tests now exercise the real CloudNotificationPlacementResolver, including machine separation, strict terminal-to-remote-workspace fallback, machine-level rows, closed/unopened workspaces and catalog projection order. The bounds regression feeds a real Pipe through CloudLinkPipe.lines: a >4 MiB line is discarded, the following 3 MiB frame survives whole, and EOF flushes the tail. No production visibility or test-only seam was added. Swift parse passes; full CI must compile and execute these now-active tests.

**Comment by teamleaderleo** [comment]
Current-head validation settlement: 3ae9b563c467cb04378b357da035b6ba57750678.

18 focused Python wiring tests pass; package/static gates and complete app+test compile admission job106126753121 pass. The newly wired notification/pipe tests compile against real current owners, without a test-only production seam.

Runtime shards have started; their result is not yet established. Shared unrelated runtime failures and runner DNS evidence are tracked centrally in #13263 (https://github.com/manaflow-ai/cmux/pull/13263#issuecomment-5751808595). No required gate has been bypassed and no blind full rerun is requested.
